UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Animation.cs

```diff
@@ -0,0 +1,335 @@
+				keyIndex => GetRightHandedQuaternion(new Quaternion(curveX.keys[keyIndex].outTangent, curveY.keys[keyIndex].outTangent, curveZ.keys[keyIndex].outTangent, curveW.keys[keyIndex].outTangent)),
+				time => GetRightHandedQuaternion(CreateNormalizedQuaternion(curveX.Evaluate(time), curveY.Evaluate(time), curveZ.Evaluate(time), curveW.Evaluate(time))),
+				values => this.ExportData(values));
+		}
+
+		private AnimationSampler ExportAnimationSamplerScale(AnimationCurve curveX, AnimationCurve curveY, AnimationCurve curveZ)
+		{
+			return this.ExportAnimationSampler(
+				new[] { curveX, curveY, curveZ },
+				keyIndex => new Vector3(curveX.keys[keyIndex].inTangent, curveY.keys[keyIndex].inTangent, curveZ.keys[keyIndex].inTangent),
+				keyIndex => new Vector3(curveX.keys[keyIndex].value, curveY.keys[keyIndex].value, curveZ.keys[keyIndex].value),
+				keyIndex => new Vector3(curveX.keys[keyIndex].outTangent, curveY.keys[keyIndex].outTangent, curveZ.keys[keyIndex].outTangent),
+				time => new Vector3(curveX.Evaluate(time), curveY.Evaluate(time), curveZ.Evaluate(time)),
+				values => this.ExportData(values));
+		}
+
+		private AnimationSampler ExportAnimationSamplerWeight(IEnumerable<AnimationCurve> weightCurves)
+		{
+			return this.ExportAnimationSampler(
+				weightCurves,
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].inTangent / 100),
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].value / 100),
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].outTangent / 100),
+				time => weightCurves.Select(curve => curve.Evaluate(time) / 100),
+				values => this.ExportData(values.SelectMany(value => value)));
+		}
+
+		private AnimationSampler ExportAnimationSampler<T>(IEnumerable<AnimationCurve> curves, Func<int, T> getInTangent, Func<int, T> getValue, Func<int, T> getOutTangent, Func<float, T> evaluate, Func<IEnumerable<T>, AccessorId> exportData)
+		{
+			//decide if we should include baking animations or not
+			if (/*bakeAnimations is False &&*/ CanExportCurvesAsSpline(curves))
+			{
+				var firstCurve = curves.First();
+
+				var input = new float[firstCurve.keys.Length];
+				var output = new T[firstCurve.keys.Length * 3];
+				for (int keyIndex = 0; keyIndex < firstCurve.keys.Length; keyIndex++)
+				{
+					input[keyIndex] = firstCurve.keys[keyIndex].time;
+
+					output[keyIndex * 3 + 0] = getInTangent(keyIndex);
+					output[keyIndex * 3 + 1] = getValue(keyIndex);
+					output[keyIndex * 3 + 2] = getOutTangent(keyIndex);
+				}
+
+				return new AnimationSampler
+				{
+					Input = this.ExportData(input, minMax: true),
+					Interpolation = GLTF.Schema.InterpolationType.CUBICSPLINE,
+					Output = exportData(output),
+				};
+			}
+			else
+			{
+				var input = new List<float>();
+				var output = new List<T>();
+				var maxTime = curves.Max(curve => curve.keys.Last().time);
+				for (float time = 0; time < maxTime; time += 1.0f / 30.0f)
```

Comment:
this should be a constant. Why 30 ms?

Reply:
I mentioned this above. It's arbitrarily set at 30fps right now. Maybe this should be a setting.